src/migrator/async.ts

```diff
@@ -28,6 +29,33 @@ import {
 	BaseAsyncMigration,
 } from './utils';
 
+const ASYNC_MIGRATION_ENV_VAR_NAME = 'PINEJS_ASYNC_MIGRATION_ENABLED';
+// for better readability of logging
+const booleanToEnabledString = (input: boolean) =>
+	input ? 'enabled' : 'disabled';
+
+let asyncMigrationIsEnabled = boolVar(ASYNC_MIGRATION_ENV_VAR_NAME, true);
+
+// log the startup condition
+(sbvrUtils.api?.migrations?.logger.info ?? console.info)(
+	`Async migration execution is ${booleanToEnabledString(
+		asyncMigrationIsEnabled,
+	)} by environment variable ${ASYNC_MIGRATION_ENV_VAR_NAME}`,
+);
+
+/**
+ * For runtime switching the async migration execution the SIGUSR2 signal is interpreted as toggle.
+ * When the process receives a SIGUSR2 signal the async migrations will toggle to be enabled or disabled.
+ */
+process.on('SIGUSR2', () => {
+	(sbvrUtils.api.migrations?.logger.info ?? console.info)(
+		`Received SIGUSR2 to toggle async migration execution from ${booleanToEnabledString(
+			asyncMigrationIsEnabled,
+		)} to ${booleanToEnabledString(!asyncMigrationIsEnabled)}`,
+	);
+	asyncMigrationIsEnabled = !asyncMigrationIsEnabled;
+});
```

Comment:
Can you instead export the ability to toggle this so that users of pinejs can choose whatever toggle mechanism they want, hardcoding SIGUSR2 into pinejs would stop any users of pinejs using that for other purposes

Reply:
@Page- 
Okay so you mean that it's exported and the pinejs user will call this with a signal string 'SIGWHATEVER' or would never call it and then it's not listening to a signal at all.

Reply:
As the migrator configuration is already exposed through pine.env.migrator I'll follow this and let the user of pinejs toggle the 
pine.env.migrator. asyncMigrationIsEnabled on their behalf. If they use a SIGUSR2 or whatever logic to toggle it.
WDYT @Page-